generic.json

```diff
@@ -0,0 +1,65 @@
+{
```

Comment:
Functionality wise, this seems pretty good ! Good job! 

It's also probably the most not-repeating you can make it. The question then becomes: is this very readable? :)

You might want to sacrifice the repetition of jsonrpc: 2.0 across a few files in favour of being able to write:

```
 	"oneOf": [
 						{ "$ref": "method/subscribe.json" },
 						{ "$ref": "method/unsubscribe.json" },
 						{ "$ref": "method/catchup.json" },
 						{ "$ref": "method/broadcast.json" },
 						{ "$ref": "method/publish.json" },
 						{ "$ref": "answer/generic.json" }
 			]
```

This implies moving the answer to a separate file, sure, but this way, you'll have a good balance of simplicity, expressiveness and non-repetition.

Side note: you might want to validate across both back-end and both front-end teams the support for relative paths in `$ref` fields in your JSON-Schema libraries. It should work seamlessly, it's more of a precaution: it would be a pity to have to go all over them again :-)

Reply:
That's a good point! I think I have a way of doing it so that the repetition of jsonrpc: 2.0 will only be in two distinct files while still keeping the directory based-structure we were talking about a few days ago (@haoqianzhang).

I will keep you updated!